proposals/mmmm-spawn-let.md

```diff
@@ -0,0 +1,701 @@
+- execute child-tasks concurrently however _not_ in parallel with the creating task (e.g. make child tasks run on the same serial executor as the calling actor),
+- if the child-task work is known to be heavy and blocking, it may be beneficial to delegate it to a specific "blocking executor" which would have a dedicated, small, number of threads on which it would execute the blocking work; Thanks to such separation, the main global thread-pool would not be impacted by starvation issues which such blocking tasks would otherwise cause.
+- various other examples where tight control over the execution context is required...
+
+We should be able to allow such configuration based on scope, like this:
+
+```swift
+await withTask(executor: .globalConcurrentExecutor) { 
+  spawn let x = ...
+  spawn let y = ...
+  // x and y execute in parallel; this is equal to the default semantics
+}
+
+actor Worker {
+  func work(first: Work, second: Work) async {
+    await withTask(executor: self.serialExecutor) {
+      // using any serial executor, will cause the tasks to be completed one-by-one,
+      // concurrently, however without any real parallelism.
+      spawn let x = process(first)
+      spawn let y = process(second)
+      // x and y do NOT execute in parallel
+    }
+  }
+}
+```
+
+The details of the API remain to be seen, but the general ability to specify an executor for child-tasks is useful and will be considered in the future.
+
+## Alternatives considered
+
+### Alternative name: `async let`
+
+The feature was previously known as `async let` yet as the work on Swift concurrency continued, based on community feedback as well as our own findings with regards of implied meanings of specific words used throughout all of Swift's concurrency APIs, we found that the word `async` is not quite right for this feature.
+
+As `spawn let` should be thought of a specialization of TaskGroups and the child-tasks spawned by either of them inherit the same kind of semantics from their enclosing parent task, it makes sense to share the same name across the two features.
+
+We feel that using the word `spawn` in _every_ case that involves the creation of a child-task is simple to understand and allows developers to learn and assume about both features in tandem. 
+
+### Explicit futures
+
+As discussed in the [structured concurrency proposal](nnnn-structured-concurrency.md#Prominent-futures), we choose not to expose futures or `Task.Handle`s for child tasks in task groups, because doing so either can undermine the hierarchy of tasks, by escaping from their parent task group and being awaited on indefinitely later, or would result in there being two kinds of future, one of which dynamically asserts that it's only used within the task's scope. `spawn let` allows for future-like data flow from child tasks to parent, without the need for general-purpose futures to be exposed.
```

Comment:
```suggestion
As discussed in the [structured concurrency proposal](0304-structured-concurrency.md#prominent-futures), we choose not to expose futures or `Task.Handle`s for child tasks in task groups, because doing so either can undermine the hierarchy of tasks, by escaping from their parent task group and being awaited on indefinitely later, or would result in there being two kinds of future, one of which dynamically asserts that it's only used within the task's scope. `spawn let` allows for future-like data flow from child tasks to parent, without the need for general-purpose futures to be exposed.
```